T/TDLib/build_tarballs.jl

```diff
@@ -0,0 +1,46 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "TDLib"
+version = v"1.6.10"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/tdlib/td.git", "78e45da6b9014f8efd96cee0ad5701464d911cc7")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+apk add gperf
+cd td/
+install_license LICENSE_1_0.txt
+sed -i 's|TD_HAS_MMSG 1|TD_HAS_MMSG 0|' tdutils/td/utils/port/config.h  # otherwise not compatible with old glibc
+mkdir build
+cd build
+cmake -DCMAKE_BUILD_TYPE=Release -DZLIB_LIBRARY=$libdir -DZLIB_INCLUDE_DIR=$includedir ..
+for f in $(grep -l $(find . -name link.txt) -e 'opt/bin/c++')
+do
+    # without this flag it cannot find libz
+    sed -i ' 1 s/$/ -lz/' $f
+done
+cmake --build . -j${nproc}
+cp libtdjson.so $libdir
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = filter(p -> Sys.islinux(p) && (libc(p) == "glibc" || arch(p) == "x86_64"), supported_platforms())
```

Comment:
The problem is that the build system is trying to run a foreign executable.  I have a small hope we can avoid that, but I don't have the time to look into it this week

Reply:
Just to understand, do you want to try to build for other platforms or happy to support only a few of them?

Reply:
I'm not familiar with such details unfortunately. Custom flags in this script come from just googling specific compilation errors and applying fixes I found somewhere.